modules/branch-keystore-node/src/branch_keystore_helpers.ts

```diff
@@ -0,0 +1,361 @@
+        typeof item[BRANCH_KEY_ACTIVE_VERSION_FIELD] === 'string',
+      `Branch keystore record does not contain a ${BRANCH_KEY_ACTIVE_VERSION_FIELD} field of type string`
+    )
+  }
+
+  //= aws-encryption-sdk-specification/framework/key-store/dynamodb-key-storage.md#record-format
+  //# 1. `enc` : Encrypted version of the key;
+  //#   represented as [AWS DDB Binary](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/HowItWorks.NamingRulesDataTypes.html#HowItWorks.DataTypes)
+  needs(
+    BRANCH_KEY_FIELD in item && item[BRANCH_KEY_FIELD] instanceof Uint8Array,
+    `Branch keystore record does not contain ${BRANCH_KEY_FIELD} field of type Uint8Array`
+  )
+
+  //= aws-encryption-sdk-specification/framework/key-store/dynamodb-key-storage.md#record-format
+  //# 1. `kms-arn`: The AWS KMS Key ARN used to generate the `enc` value.
+  //#   represented as [AWS DDB String](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/HowItWorks.NamingRulesDataTypes.html#HowItWorks.DataTypes)
+  needs(
+    KMS_FIELD in item && typeof item[KMS_FIELD] === 'string',
+    `Branch keystore record does not contain ${KMS_FIELD} field of type string`
+  )
+
+  //= aws-encryption-sdk-specification/framework/key-store/dynamodb-key-storage.md#record-format
+  //# 1. `create-time`: Timestamp in ISO 8601 format in UTC, to microsecond precision.
+  //#   Represented as [AWS DDB String](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/HowItWorks.NamingRulesDataTypes.html#HowItWorks.DataTypes)
+  needs(
+    KEY_CREATE_TIME_FIELD in item &&
+      typeof item[KEY_CREATE_TIME_FIELD] === 'string',
+    `Branch keystore record does not contain ${KEY_CREATE_TIME_FIELD} field of type string`
+  )
+
+  //= aws-encryption-sdk-specification/framework/key-store/dynamodb-key-storage.md#record-format
+  //# 1. `hierarchy-version`: Version of the hierarchical keyring;
+  //#   represented as [AWS DDB Number](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/HowItWorks.NamingRulesDataTypes.html#HowItWorks.DataTypes)
+  needs(
+    HIERARCHY_VERSION_FIELD in item &&
+      typeof item[HIERARCHY_VERSION_FIELD] === 'number',
+    `Branch keystore record does not contain ${HIERARCHY_VERSION_FIELD} field of type number`
+  )
+
+  //= aws-encryption-sdk-specification/framework/key-store/dynamodb-key-storage.md#record-format
+  //# A branch key record MAY include [custom encryption context](../branch-key-store.md#custom-encryption-context) key-value pairs.
+  //# These attributes should be prefixed with `aws-crypto-ec:` the same way they are for [AWS KMS encryption context](../branch-key-store.md#encryption-context).
+  for (const field in item) {
+    if (!POTENTIAL_BRANCH_KEY_RECORD_FIELDS.includes(field)) {
+      needs(
+        field.startsWith(CUSTOM_ENCRYPTION_CONTEXT_FIELD_PREFIX),
+        `Custom encryption context key ${field} should be prefixed with ${CUSTOM_ENCRYPTION_CONTEXT_FIELD_PREFIX}`
```

Comment:
If I am reading this correctly, the other implementation is not as strict in this respect. 

I'm not against it, but I want to call out the diff. 

I.e: other languages would allow for un-prefixed values.